Sources/ScriptingCore/utils.swift

```diff
@@ -0,0 +1,121 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright 2015 - 2021 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+
+import Basics
+import Foundation
+import ScriptParse
+import TSCBasic
+
+fileprivate extension AbsolutePath {
+    var sha256Hash: String {
+        String(ByteString(stringLiteral: pathString)
+                .sha256Checksum.prefix(6))
+    }
+}
+
+/// An enumeration of the errors that can be generated by the run tool.
+public enum ScriptError: Swift.Error {
+    /// The specified file doesn't exist.
+    case fileNotFound(String)
+    /// The target path is directory
+    case isDirectory(String)
```

Comment:
Should these two maybe store `AbsolutePath`s?

Reply:
These may receive user inputs that is even not a valid path. `String` is better as they had no meanings but for display use.